learningresources/models.py

```diff
@@ -228,3 +202,46 @@ class Meta:  # pylint: disable=missing-docstring
             RepoPermission.add_edit_metadata,
             RepoPermission.manage_repo_users,
         )
+
+
+def get_preview_url(resource, org=None, course_number=None, run=None):
```

Comment:
I had to take it out of the model because it caused problems during migrations -- the `LearningResource` in a migration, from the migrations ORM, doesn't have the `get_preview_url` method.

However, I didn't see any reason to move it to another file.